custom_components/adaptive_lighting/switch.py

```diff
@@ -838,6 +841,11 @@ def __init__(
             data,
         )
 
+    def __getattribute__(self, name):
```

Comment:
what about creating a property on the switch:
```
@property
def all_lights(self):
    return self._expand_light_groups()
```

then use `switch.all_lights` everywhere?

Reply:
brilliant.

Reply:
Done